k3k-kubelet/kubelet.go

```diff
@@ -109,24 +115,50 @@ func newKubelet(ctx context.Context, c *config, logger *k3klog.Logger) (*kubelet
 		},
 	})
 
+	logger.Info("adding service syncer controller")
+	if k3kkubeletcontroller.AddServiceSyncer(ctx, virtualMgr, hostMgr, c.ClusterName, c.ClusterNamespace, k3klog.New(false)); err != nil {
+		return nil, fmt.Errorf("failed to add service syncer controller: %v", err)
+	}
+
+	clusterIP, err := clusterIP(ctx, c.AgentHostname, c.ClusterNamespace, hostClient)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to extract the clusterIP for the server service %s", err.Error())
```

Comment:
Let's use errors.New and the + operator here and anywhere we're concatenating strings.